### PR TITLE
String reverse

### PR DESCRIPTION
**Pull request by Hargeon**
Added tests and implemented Reverse function

### CODE REVIEW

File: strings_helper.go [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+	reversedStr := make([]rune, strLen)
+	charIndex := strLen - 1
+	for _, char := range str {
+		reversedStr[charIndex] = char
+		charIndex--
+	}
+	return string(reversedStr)
+}
+
+// CharNumber returns a number of characters in the string
```
Comment (gordonBusyman) [inline strings_helper.go:20]: This function is not required for the first release

File: strings_helper.go [hunk 2]
```diff
@@ -0,0 +1,29 @@
+package strings_helper
+
+import (
+	"unicode"
+	"unicode/utf8"
+)
+
+// Reverse string
```
Comment (gordonBusyman) [inline strings_helper.go:8]: Can we a add a more descriptive comment

File: strings_helper.go [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+func Reverse(str string) string {
```
Comment (gordonBusyman) [inline strings_helper.go:9]: Use new lines to separate the logical code blocks if you _want_

File: strings_helper_test.go [hunk 4]
```diff
@@ -0,0 +1,82 @@
+package strings_helper
+
+import (
+	"fmt"
+	"testing"
+)
+
+func TestReverse(t *testing.T) {
```
Comment (gordonBusyman) [inline strings_helper_test.go:8]: Add `t.Run()` for test cases with the name of the test (what is being tested)

File: strings_helper_test.go [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+			result: "тевирП",
+		},
+		{
+			input:  "A",
+			result: "A",
+		},
+		{
+			input:  "",
+			result: "",
+		},
```
Comment (gordonBusyman) [inline strings_helper_test.go:28]: Add one more test to check the phrase, ex: "Мама мыла раму"

File: strings_helper_test.go [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+	}
+
+	for _, testCase := range cases {
+		result := Reverse(testCase.input)
+		if result != testCase.result {
+			t.Errorf("For case %v expected: %v, got: %v\n", testCase.input, result, testCase.result)
```
Comment (gordonBusyman) [inline strings_helper_test.go:34]: In this case formatter %q  is more appropriate, ex.: `For case "" expected: "", got: "r"` is the message for failed the case [here](https://github.com/Hargeon/strings_helper/blob/e3c9828853a42f5181edd85c1db689612650f781/strings_helper_test.go#L25)

**Review by gordonBusyman — CHANGES_REQUESTED** [review]
Looks good! 
Please address the comments. Also I highly recommend to rename the pkg

File: .idea/.gitignore [hunk 7]
```diff
@@ -0,0 +1 @@
+.idea
```
[the remark this answers is not in the document]
Comment (Hargeon) [reply .idea/.gitignore:1]: Fixed in new commit

File: README.md [hunk 8]
```diff
@@ -1 +1,10 @@
-# strings_helper
\ No newline at end of file
+Task:
```
[the remark this answers is not in the document]
Comment (Hargeon) [reply README.md]: Fixed in new commit

File: go.mod [hunk 9]
```diff
@@ -0,0 +1 @@
+module "github.com/Hargeon/strings_helper"
```
[the remark this answers is not in the document]
Comment (Hargeon) [reply go.mod]: Fixed in new commit

File: strings_helper.go [hunk 10]
```diff
@@ -0,0 +1,29 @@
+package strings_helper
```
[the remark this answers is not in the document]
Comment (Hargeon) [reply strings_helper.go]: Fixed in new commit

File: strings_helper.go (hunk 2 above)
Comment (Hargeon) [reply strings_helper.go]: Fixed in new commit

File: strings_helper.go (hunk 3 above)
Comment (Hargeon) [reply strings_helper.go:10]: Fixed in new commit

File: strings_helper_test.go (hunk 4 above)
Comment (Hargeon) [reply strings_helper_test.go:8]: Fixed in new commit

File: strings_helper_test.go (hunk 5 above)
Comment (Hargeon) [reply strings_helper_test.go:32]: Fixed in new commit

File: strings_helper_test.go (hunk 6 above)
Comment (Hargeon) [reply strings_helper_test.go]: Fixed in new commit

File: .idea/.gitignore (hunk 7 above)
[the remark this answers is not in the document]
Comment (gordonBusyman) [reply .idea/.gitignore:1]: Please remove this

File: README.md [hunk 11]
```diff
@@ -1 +1,34 @@
-# strings_helper
\ No newline at end of file
+# stringhlp - helper for string manipulation
```
Comment (gordonBusyman) [inline README.md:1]: Looks good^ but please add more details described in https://docs.github.com/en/github/creating-cloning-and-archiving-repositories/creating-a-repository-on-github/about-readmes

File: strings_helper.go (hunk 1 above)
Comment (gordonBusyman) [reply strings_helper.go:23]: Please pay attention to this comment

File: strings_helper_test.go [hunk 12]
```diff
@@ -0,0 +1,114 @@
...
+			result: "esarhp gib yreV",
+		},
+		{
+			input:  "gnirts desreveR",
+			result: "Reversed string",
+		},
+	}
+
+	for _, testCase := range cases {
+		t.Run(testCase.input, testReverseFunc(testCase.input, testCase.result))
```
Comment (gordonBusyman) [inline strings_helper_test.go:44]: Usually for the name of the test case we use another property "name", like:
```
...
		{
                        name: "None ASCII phrase",
			input:  "Мама мыла раму",
			result: "умар алым амаМ",
		},
...
t.Run(testCase.name, testReverseFunc(testCase.input, testCase.result))
...
```

File: strings_helper_test.go [hunk 13]
```diff
@@ -0,0 +1,114 @@
...
+		}
+	}
+}
+
+func ExampleReverse() {
+	testStr := "Привет"
+	fmt.Println(Reverse(testStr))
+	// Output:
+	// тевирП
+}
```
Comment (gordonBusyman) [inline strings_helper_test.go:107]: This is the example that can be moved to README _(as you already have done)_

File: .idea/.gitignore (hunk 7 above)
[the remark this answers is not in the document]
Comment (Hargeon) [reply .idea/.gitignore]: Done

File: strings_helper.go (hunk 1 above)
Comment (Hargeon) [reply strings_helper.go:22]: Commented the method

File: README.md (hunk 11 above)
Comment (Hargeon) [reply README.md]: Done

File: strings_helper_test.go (hunk 12 above)
Comment (Hargeon) [reply strings_helper_test.go]: Done

File: strings_helper_test.go (hunk 13 above)
Comment (Hargeon) [reply strings_helper_test.go]: Done

File: .gitignore [hunk 14]
```diff
@@ -0,0 +1 @@
+.idea
```
Comment (gordonBusyman) [inline .gitignore:1]: [Here](https://github.com/github/gitignore) you can find examples of correct `.gitignore` files

File: README.md [hunk 15]
```diff
@@ -1 +1,33 @@
-# strings_helper
\ No newline at end of file
+# stringhlp
+
+A Go Package That Makes Strings Easier.
+
+## Installation
+```bash
+go get -u github.com/Hargeon/stringhlp
```
Comment (gordonBusyman) [inline README.md:7]: What is the `-u` flag for?

**Comment by gordonBusyman** [comment]
Please remove the `./idea` folder - it is empty

**Review by gordonBusyman — CHANGES_REQUESTED** [review]
Left final comments. This is it and we are gtg

File: .gitignore (hunk 14 above)
Comment (Hargeon) [reply .gitignore:2]: Done

File: README.md (hunk 15 above)
Comment (Hargeon) [reply README.md:7]: The flag helps to upgrade to a new minor version if someone is already using the old one

**Review by gordonBusyman — APPROVED** [review]

**Pull request merged**